### PR TITLE
(LD-163) develop restaurant details view + remove unneeded text

### PR DESCRIPTION
**Pull request by snoattoh**
- Updates the restaurant details view to implement the useRestaurant hook
- Adds testing for restaurant details
- removes unneeded text on preceding views
- updates text on preceding views

I'll fix up the colors throughout the application in a follow up styling PR.

I'll be addressing the [TableView](https://developer.apple.com/documentation/uikit/views_and_controls/table_views) update in another PR.

### CODE REVIEW

File: src/components/RestaurantHeader/RestaurantHeader.tsx [hunk 1]
```diff
@@ -3,29 +3,31 @@ import type { Restaurant } from "../../services/restaurant"
...
-}) => {
+type Props = {
+  restaurant: Restaurant | null
+}
+
+const RestaurantHeader: React.FC<Props> = ({ restaurant }) => {
   return (
     <View style={styles.restaurantHeader}>
-      <ImageBackground source={{ uri: `url(/${restaurant.images.banner})` }}>
+      <ImageBackground source={{ uri: `${restaurant?.images.banner}` }}>
```
Comment (chasenlehara) [inline src/components/RestaurantHeader/RestaurantHeader.tsx]: Is this working? I don’t see it in the screenshots.

File: src/screens/RestaurantDetails/RestaurantDetails.test.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import RestaurantDetails from "./RestaurantDetails"
+import { NavigationContainer } from "@react-navigation/native"
+
+// Mock the hooks and components used in RestaurantDetails
+
+let useRestaurant: jest.SpyInstance<ReturnType<typeof hook.useRestaurant>>
+beforeEach(() => {
+  jest.resetAllMocks()
+  useRestaurant = jest.spyOn(hook, "useRestaurant")
+})
```
Comment (chasenlehara) [inline src/screens/RestaurantDetails/RestaurantDetails.test.tsx]: I think it’s more common to have these blocks inside the `describe`.

File: src/screens/RestaurantDetails/RestaurantDetails.test.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    )
+    expect(
+      screen.getByText(/Error loading restaurant details:/i, {
+        exact: false,
+      }),
+    ).toBeOnTheScreen()
+    expect(screen.getByText(/Mock error/i)).toBeOnTheScreen()
+  })
+
+  it("renders the RestaurantHeader and content when data is available", () => {
```
Comment (chasenlehara) [inline src/screens/RestaurantDetails/RestaurantDetails.test.tsx:62]: The “when data is available” makes me think there should be a matching “when data is not available.”

File: src/screens/RestaurantList/RestaurantList.tsx [hunk 4]
```diff
@@ -18,54 +17,49 @@ const RestaurantList: FC<Props> = ({ route }) => {
...
+    <Box padding="s" style={styles.container}>
+      <FlatList
+        data={data}
+        renderItem={({ item: restaurant }) => (
+          <Press
+            title={restaurant.name}
+            onPress={() =>
+              navigation.navigate("RestaurantDetails", {
+                slug: restaurant.slug,
+              })
```
Comment (chasenlehara) [inline src/screens/RestaurantList/RestaurantList.tsx:47]: In the future, let’s move these callbacks out of the JSX (into a function like `handlePress`).

**Review by chasenlehara — APPROVED** [review]
It looks like there are a number of changes without associated test file changes, so be sure to add basic tests in a follow-up PR.

File: src/components/RestaurantHeader/RestaurantHeader.tsx (hunk 1 above)
Comment (snoattoh) [reply src/components/RestaurantHeader/RestaurantHeader.tsx]: Thank again for taking the time to help with sorting this out.

**Pull request merged**